hazelcast/src/main/java/com/hazelcast/internal/monitors/MultiLinePerformanceLogWriter.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.monitors;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+import static java.util.Calendar.DAY_OF_MONTH;
+import static java.util.Calendar.HOUR;
+import static java.util.Calendar.MINUTE;
+import static java.util.Calendar.MONTH;
+import static java.util.Calendar.SECOND;
+import static java.util.Calendar.YEAR;
+
+/**
+ * A {@link PerformanceLogWriter} that writes over multiple lines. Useful for human reading.
+ */
+public class MultiLinePerformanceLogWriter extends PerformanceLogWriter {
+
+    private static final String STR_LONG_MIN_VALUE = String.format("%,d", Long.MIN_VALUE);
+
+    private static final char[] DIGITS = new char[]{'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+
+    private static final String[] INDENTS = new String[]{
+            "\n                          ",
```

Comment:
A sidenote: writing `String.format("%n                              ")` takes care of the line separator, if you ever feel the need for that.